tests/unit/s2n_extension_list_parse_test.c

```diff
@@ -374,6 +380,43 @@ int main()
         EXPECT_SUCCESS(s2n_stuffer_free(&stuffer));
     }
 
+    /* Test parsed extensions assigned correct indexes */
+    {
+        s2n_parsed_extensions_list parsed_extension_list = { 0 };
+        struct s2n_stuffer stuffer = { 0 };
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&stuffer, 0));
+
+        s2n_extension_type test_extension_2 = test_extension;
+        test_extension_2.iana_value = TLS_EXTENSION_SIGNATURE_ALGORITHMS;
+        s2n_extension_type test_extension_3 = test_extension;
+        test_extension_3.iana_value = TLS_EXTENSION_ALPN;
+
+        /* Reserve size */
+        struct s2n_stuffer_reservation extension_list_size = {0};
+        EXPECT_SUCCESS(s2n_stuffer_reserve_uint16(&stuffer, &extension_list_size));
+        /* Write extensions */
+        EXPECT_SUCCESS(s2n_extension_send(&test_extension, conn, &stuffer));
+        EXPECT_SUCCESS(s2n_extension_send(&test_extension_2, conn, &stuffer));
+        EXPECT_SUCCESS(s2n_stuffer_write_uint16(&stuffer, S2N_UNKNOWN_EXTENSION_IANA));
+        EXPECT_SUCCESS(s2n_stuffer_write_uint16(&stuffer, 0));
+        EXPECT_SUCCESS(s2n_extension_send(&test_extension_3, conn, &stuffer));
+        /* Check / write size */
+        EXPECT_TRUE(s2n_stuffer_data_available(&stuffer) > extension_list_size.length);
+        EXPECT_SUCCESS(s2n_stuffer_write_vector_size(&extension_list_size));
+
+        EXPECT_SUCCESS(s2n_extension_list_parse(&stuffer, &parsed_extension_list));
+
+        uint16_t expected_order[] = { test_extension.iana_value, test_extension_2.iana_value, test_extension_3.iana_value };
+        for (size_t i = 0; i < s2n_array_len(expected_order); i++) {
+            s2n_extension_type_id id;
+            EXPECT_SUCCESS(s2n_extension_supported_iana_value_to_id(expected_order[i], &id));
+            EXPECT_EQUAL(parsed_extension_list.parsed_extensions[id].wire_index, i);
+        }
+        EXPECT_EQUAL(parsed_extension_list.count, 3);
```

Comment:
Can we add a test to ensure that we donot exceed the upper limit for `parsed_extension_list.count`?

Reply:
How would you set up that test? The only limit on parsed_extension_list.count is the size of uint16_t, and you can't receive more extensions than that because the size of the extension list on the wire is also uint16_t.

Reply:
We could probably check for max value being not more than size of uint16_t then :-)

Reply:
It can't be larger than uint16_t, because both .count and the size of the extension list are uint16_ts.